pkg/server/api/shim.go

```diff
@@ -120,6 +120,8 @@ func shimConfig(cniConfig []byte) (*ShimNetConf, error) {
 	}
 	if multusConfig.MultusSocketDir == "" {
 		multusConfig.MultusSocketDir = defaultMultusRunDir
+	} else {
+		multusConfig.MultusSocketDir = strings.ReplaceAll(multusConfig.MultusSocketDir, "/host", "")
```

Comment:
@dougbtv / @s1061123 this is the most fishy part in this PR IMO ...

Seems we've always had a bug when specifying the socket path - from the daemon's perspective, the socket will be in `/host/<whatever>` while from the shim's perspective it'll be in `<whatever>` . 

... since we were relying on the defaults (and the shim / daemon used different defaults) we were not seeing it.

I don't have a smarter idea right now on how to achieve this to be honest. Providing a new know for the socket path from the shim's perspective seems weird. And keep in mind this must be configured via the volume / hostPath mapping as well, so, the configuration must be aligned.

Reply:
Your code change seems to make consistency between shim's socket parameter and multus-daemon's socket parameter, however, I believe that we should explicitly have shim's socket directory option and multus-daemon's socket directory as separate item.

The reason is user may change multus-daemonset.yml manifest and change '/host' prefix. At present, multus-daemon   could be still run as host process (just unix-daemon). So your code implicitly assumes that multus-daemon must runs in a pod always hence multus-daemon no longer run as host process. In another case, if user wants to change host prefix from '/host' to something different, then it may be broken.

For now, we should expect that if some user want to change these parameter, then they should understand what the thick plugin and shim plugin are and they should make consistency between multus-daemon's socket path and multus-shim's socket path manually. I suppose it should make sense (even though it is not good UI).

From multus-shim and multus-daemon component point of view, we should not have such verification I believe.
If we need to do that, this could be done in manifest orchestration or some manifest verification , I suppose.

Reply:
Makes sense; I understand my current proposal is pretty bad, for the reasons you state.

... but it's good at showing something's wrong :) 

How should I call the parameters ? serverSocketDir / daemonSocketDir and clientSocketDir / shimSocketDir ? 

I would go with daemonSocketDir and shimSocketDir. Any objections @s1061123 ?

Reply:
daemonSocketDir and shimSocketDir are good name, thanks!

Reply:
+1 on the naming

Reply:
Done.